flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/Executing.java

```diff
@@ -60,8 +69,16 @@
         this.context = context;
         this.userCodeClassLoader = userCodeClassLoader;
 
-        deploy();
-
+        if (executingStateBehavior == Behavior.DEPLOY_ON_ENTER) {
+            deploy();
+        } else if (executingStateBehavior == Behavior.EXPECT_RUNNING) {
+            Preconditions.checkState(
+                    executionGraph.getState() == JobStatus.RUNNING,
+                    "Assuming running execution graph");
+        } else {
+            throw new IllegalStateException(
+                    "Unexpected executing state behavior " + executingStateBehavior);
+        }
```

Comment:
Having slept over this one night, I'm not so sure anymore if this is the right approach. We can probably always assume the execution graph to be in state RUNNING, and on `Behavior.EXPECT_RUNNING` we can go through all ExecutionVertex and check if their state is running. I'll try to look into this Monday morning latest.

Reply:
I changed it.